pan-domain-auth-core/src/main/scala/com/gu/pandomainauth/PanDomainAuthSettingsRefresher.scala

```diff
@@ -1,83 +1,61 @@
 package com.gu.pandomainauth
 
-import com.amazonaws.auth.{DefaultAWSCredentialsProviderChain, AWSCredentialsProvider}
-import com.amazonaws.regions.{Regions, Region}
+import java.util.concurrent.atomic.AtomicReference
+import java.util.concurrent.{Executors, ScheduledExecutorService, TimeUnit}
 
-import scala.concurrent.ExecutionContext.Implicits.global
-import scala.concurrent.duration._
-import scala.language.postfixOps
-import akka.actor.{Props, Actor, ActorSystem}
-import akka.agent.Agent
-import akka.event.Logging
+import com.amazonaws.services.s3.AmazonS3
 import com.gu.pandomainauth.model.PanDomainAuthSettings
-import com.gu.pandomainauth.service.{ProxyConfiguration, S3Bucket}
+import org.slf4j.LoggerFactory
 
-import scala.concurrent.duration.FiniteDuration
+import scala.language.postfixOps
 
 /**
   * PanDomainAuthSettingsRefresher will periodically refresh the pan domain settings and expose them via the "settings" method
   *
-  * @param domain the domain you are authin agains
+  * @param domain the domain you are authenticating against
   * @param system the identifier for your app, typically the same as the subdomain your app runs on
   * @param bucketName the bucket where the settings are stored
-  * @param actorSystem the actor system to create the refresh actor
-  * @param awsCredentialsProvider AWS credential provider
-  * @param awsRegion AWS region
-  * @param proxyConfiguration optional proxy configuration
+  * @param settingsFileKey the name of the file that contains the private settings for the given domain
+  * @param s3Client the AWS S3 client that will be used to download the settings from the bucket
+  * @param scheduler optional scheduler that will be used to run the code that updates the bucket
   */
 class PanDomainAuthSettingsRefresher(
   val domain: String,
   val system: String,
-  bucketName: String,
-  actorSystem: ActorSystem,
-  awsCredentialsProvider: AWSCredentialsProvider,
-  awsRegion: Regions,
-  proxyConfiguration: Option[ProxyConfiguration] = None
+  val bucketName: String,
+  settingsFileKey: String,
+  val s3Client: AmazonS3,
```

Comment:
This change to simply take an existing S3 client means we have much more scope for integrating with other environments. It also means we no longer need the credentials provider, region or proxy configuration since they are set on the builder used to create the client (ie in calling code)